redish/libs/redish-frontend/authentication/data-access/src/lib/facade/authentication.facade.ts

```diff
@@ -0,0 +1,149 @@
+ */
+export function useAuthenticationFacade(
+  setToken: (token: string | null) => void
+): {
+  registerViewModel: {
+    registerUser: RegisterUser;
+    handleUsernameChanged: (username: string) => void;
+    handleEmailChanged: (email: string) => void;
+    handlePasswordChanged: (password: string) => void;
+    handleSubmit: () => Promise<void>;
+  };
+  loginViewModel: {
+    loginUser: LoginUser;
+    handleEmailChanged: (email: string) => void;
+    handlePasswordChanged: (password: string) => void;
+    handleSubmit: () => Promise<void>;
+  };
+} {
+  /**
+   * REGISTER
+   */
+  const initialRegisterUser: RegisterUser = {
+    username: '',
+    password: '',
+    email: '',
+  };
+
+  const [registerUser, updateRegisterUser] =
+    useImmer<RegisterUser>(initialRegisterUser);
+
+  const handleRegisterUserUsernameChanged = (username: string) => {
+    updateRegisterUser((draft) => {
+      draft.username = username;
+    });
+  };
+
+  function handleRegisterUserEmailChanged(email: string) {
+    updateRegisterUser((draft) => {
+      draft.email = email;
+    });
+  }
+
+  function handleRegisterUserPasswordChanged(password: string) {
+    updateRegisterUser((draft) => {
+      draft.password = password;
+    });
+  }
+
+  async function handleRegisterUserSubmit(): Promise<void> {
+    const result = await authenticationApiService.register(
+      registerUser.username,
+      registerUser.email,
+      registerUser.password
+    );
+
+    setToken(result);
+  }
+
+  /**
+   * LOGIN
+   */
+  const initialLoginUser: LoginUser = {
+    password: '',
+    email: '',
+  };
+
+  const [loginUser, updateLoginUser] = useImmer<LoginUser>(initialLoginUser);
+
+  function handleLoginUserEmailChanged(email: string) {
+    updateLoginUser((draft) => {
+      draft.email = email;
+    });
+  }
+
+  function handleLoginUserPasswordChanged(password: string) {
+    updateLoginUser((draft) => {
+      draft.password = password;
+    });
+  }
+
+  const navigate = useNavigate();
+
+  async function handleLoginUserSubmit(): Promise<void> {
+    const token = await authenticationApiService.login(
+      loginUser.email,
+      loginUser.password
+    );
+
+    if (typeof token !== 'string') {
+      return;
+    }
+
+    setToken(token);
+    navigate('/');
+  }
+
+  return {
+    registerViewModel: {
+      registerUser,
+      handleUsernameChanged: handleRegisterUserUsernameChanged,
+      handleEmailChanged: handleRegisterUserEmailChanged,
+      handlePasswordChanged: handleRegisterUserPasswordChanged,
+      handleSubmit: handleRegisterUserSubmit,
+    },
+    loginViewModel: {
+      loginUser,
+      handleEmailChanged: handleLoginUserEmailChanged,
+      handlePasswordChanged: handleLoginUserPasswordChanged,
+      handleSubmit: handleLoginUserSubmit,
+    },
+  };
+}
+
+export const useAuthenticationCore = () => {
+  // State to hold the authentication token
+  const [token, _setToken] = useState<string | null>(
```

Comment:
"useState call is not destructured into value + setter pair" sonarlint

Reply:
I think this was linted, because the function was saved in a variable instead of just declared. This way it should be fine.